app/screens/timezone/select_timezone/select_timezone.js

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    View,
+    Text,
+    FlatList,
+    Platform,
+    TouchableOpacity,
+} from 'react-native';
+import {intlShape} from 'react-intl';
+
+import CheckMark from 'app/components/checkmark';
+import SearchBar from 'app/components/search_bar';
+import StatusBar from 'app/components/status_bar';
+import {ListTypes} from 'app/constants';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import {getTimezoneRegion} from 'app/utils/timezone';
+
+const ITEM_HEIGHT = 45;
+const VIEWABILITY_CONFIG = ListTypes.VISIBILITY_CONFIG_DEFAULTS;
+
+export default class Timezone extends PureComponent {
+    static propTypes = {
+        selectedTimezone: PropTypes.string.isRequired,
+        initialScrollIndex: PropTypes.number.isRequired,
+        timezones: PropTypes.array.isRequired,
+        navigator: PropTypes.object,
+        onBack: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+    };
+
+    static contextTypes = {
+        intl: intlShape,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: '',
+            timezones: props.timezones,
+        };
+    }
+
+    filteredTimezones = (timezonePrefix) => {
+        if (timezonePrefix.length === 0) {
+            return this.state.timezones;
+        }
+
+        const lowerCasePrefix = timezonePrefix.toLowerCase();
+
+        return this.state.timezones.filter((t) => (
+            getTimezoneRegion(t).toLowerCase().indexOf(lowerCasePrefix) >= 0 ||
+            t.toLowerCase().indexOf(lowerCasePrefix) >= 0
+        ));
+    };
+
+    selectTimezone = (timezone) => {
+        this.props.onBack(timezone);
+        this.props.navigator.pop();
+    };
+
+    handleTextChanged = (value) => {
+        this.setState({value});
+    };
+
+    attachList = (l) => {
+        this.list = l;
+    };
+
+    keyExtractor = (item) => item;
+
+    getItemLayout = (data, index) => ({
+        length: ITEM_HEIGHT,
+        offset: ITEM_HEIGHT * index,
+        index,
+    });
+
+    renderItem = ({item: timezone}) => {
+        const {theme, selectedTimezone} = this.props;
+        const style = getStyleSheet(theme);
+
+        const selected = timezone === selectedTimezone && (
+            <CheckMark
+                width={12}
+                height={12}
+                color={theme.linkColor}
+            />
+        );
+
+        return (
+            <TouchableOpacity
+                style={style.itemContainer}
+                key={timezone}
+                onPress={() => this.selectTimezone(timezone)}
```

Comment:
can you please make this another component and then avoid using an arrow function in the prop

Reply:
Just curious @enahum, what do we gain from refactoring this into its own component?

Ultimately the arrow function would still be used since we depend on passing a specific timezone to the function.

If we move this into its own component we would only move this, I would say it would mildly improve readability.   Is that the concern here for the refactor? to improve readability?

```
renderItem = ({item: timezone}) => {
        const {theme, selectedTimezone} = this.props;
        const style = getStyleSheet(theme);

        const selected = timezone === selectedTimezone && (
            <CheckMark
                width={12}
                height={12}
                color={theme.linkColor}
            />
        );

        return (
            <TouchableOpacity
                style={style.itemContainer}
                key={timezone}
                onPress={() => this.selectTimezone(timezone)}
            >
                <View style={style.item}>
                    <Text style={style.itemText}>
                        {timezone}
                    </Text>
                </View>
                {selected}
            </TouchableOpacity>
        );
    };
```

Reply:
I take back

> Ultimately the arrow function would still be used since we depend on passing a specific timezone to the function.

We could leverage this.props.timezone when calling onPress in the component.

I may not recall, is the arrow function in the onPress a matter of performance degradation or readability? 😓